### PR TITLE
fix: don't return syntax error for Annotated's metadata

### LINKED ISSUE #574: syntax error in forward annotation when using Annotated
When using [`Annotated`](https://www.python.org/dev/peps/pep-0593/) with string annotations the following error is raised:

```
./pyflakes.py:3:16 syntax error in forward annotation '>1'
```

Triggering code:

```
from typing import Annotated

Annotated[int, '>1']
```

### PR DESCRIPTION
**Pull request by sk-**
PEP 593 https://www.python.org/dev/peps/pep-0593/ introduced the type Annotated, which allows to add metadata to types. By the definition only the first argument is required to be a proper type, all the other arguments are dependent on the consumer and can be other types or even literals.

This PR fixes #574.

### CODE REVIEW

**Comment by sk-** [comment]
First pass at fixing #574, this is my first contribution to `pyflakes` so let me know if any changes are required.

Note also that the failing test in Appveyor is because it failed to install `pypy`.

```
(New-Object Net.WebClient).DownloadFile('https://bitbucket.org/pypy/pypy/downloads/pypy2.7-v7.1.1-win32.zip', "$env:appveyor_build_folder\pypy2.7-v7.1.1-win32.zip")
Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."
At line:1 char:1
+ (New-Object Net.WebClient).DownloadFile('https://bitbucket.org/pypy/p ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
 
Command executed with exception: Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."
```

File: pyflakes/checker.py
Comment (asottile) [inline pyflakes/checker.py:1492]: instead of a new attribute, I think you can visit the annotation part with `_in_annotation` True and visit the non-annotation part with it `False`

File: pyflakes/test/test_type_annotations.py
Comment (asottile) [inline pyflakes/test/test_type_annotations.py:526]: these two tests are already tested above, I don't think this adds any coverage that isn't already there

File: pyflakes/test/test_type_annotations.py
Comment (asottile) [inline pyflakes/test/test_type_annotations.py:553]: the decorators on these three tests are incorrect -- there's nothing python3.6-specific about these particular tests

**Comment by asottile** [comment]
if you rebase, appveyor should be fixed

File: pyflakes/test/test_type_annotations.py
Comment (sk-) [reply pyflakes/test/test_type_annotations.py:535]: Thanks, I had recalled wrongly that this was supported only for python 3.6 onwards. But it is supported for python >= 3.4. See https://github.com/python/typing/pull/637

File: pyflakes/checker.py
Comment (sk-) [reply pyflakes/checker.py:1484]: Done

File: pyflakes/test/test_type_annotations.py
Comment (sk-) [reply pyflakes/test/test_type_annotations.py]: Thanks, I removed those tests.

File: pyflakes/checker.py
Comment (asottile) [inline pyflakes/checker.py:1496]: I believe this needs to be restored afterwards (as in a context manager that does the opposite of the one which enables `_in_annotation`)

for example this should fail (and may not with this change?):

```python
def f(x: Union[Annotated['int', '>0'], 'integer']) -> ...:
```

Comment (sk-) [reply pyflakes/checker.py]: You are right, that test case failed. I added it as a test case and fixed the code.

**Review by asottile — APPROVED** [review]

**Comment by sk-** [comment]
Thanks @asottile for the quick review.

**Review by sigmavirus24 — APPROVED** [review]

**Comment by asottile** [comment]
this has a conflict with #535 -- ideally I'd like to merge the other one first (since it was written first) and then update this PR -- thoughts?

**Comment by sigmavirus24** [comment]
@asottile that makes sense. The other one was already hit with a different merge conflict. Have you been holding off on merging that one for a reason?

**Comment by asottile** [comment]
@sigmavirus24 iirc pyflakes has 2-signoff so I was going to wait for a second review (though, given inactivity maybe we should go to 1-signoff?)

**Comment by sigmavirus24** [comment]
> iirc pyflakes has 2-signoff

If it does, it's not configured that way with the settings. I don't remember the requirement being 2 approvals, but I'll go review that one and see if I can approve it.

File: pyflakes/checker.py
Comment (asottile) [inline pyflakes/checker.py:1310]: this can reuse `_enter_annotation(ann_type=AnnotationState.NONE)`

File: pyflakes/checker.py
Comment (asottile) [inline pyflakes/checker.py:1536]: this if block ends up being not-quite-correct in python3.9+  -- the `ast.Index` type is gone and this is just a `Tuple` directly (this leads to one test failure on python3.9 -- let me know if you want me to try and hack on this for ya since it's a bit tricky)

Comment (sk-) [reply pyflakes/checker.py:1528]: Thanks @asottile for the heads up. If you could work on this that'd be great.

File: pyflakes/checker.py
Comment (sk-) [reply pyflakes/checker.py]: Done.

**Comment by asottile** [comment]
@sk-  let me know how that looks!

**Comment by sk-** [comment]
@asottile Looks great, specially the extraction of the function `_is_name_or_attr`.

**Pull request merged**
